packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -175,18 +176,22 @@ contract UniswapFeeHandlerSeller is IFeeHandlerSeller, FeeHandlerSeller {
     uint256 amount,
     IUniswapV2RouterMin bestRouter
   ) private view returns (uint256) {
-    ISortedOracles sortedOracles = getSortedOracles();
+    address _oracleAddress = getOracleAddress(sellTokenAddress);
+
     uint256 minReports = minimumReports[sellTokenAddress];
 
-    require(
-      sortedOracles.numRates(sellTokenAddress) >= minReports,
-      "Number of reports for token not enough"
-    );
+    IOracle oracle = IOracle(_oracleAddress);
 
     uint256 minimalSortedOracles = 0;
     // if minimumReports for this token is zero, assume the check is not needed
     if (minReports > 0) {
-      (uint256 rateNumerator, uint256 rateDenominator) = sortedOracles.medianRate(sellTokenAddress);
+      ISortedOracles sortedOracles = ISortedOracles(_oracleAddress);
```

Comment:
This feels a bit weird that we make an assumption that IOracle is also ISortedOracle